crates/engine/tree/src/tree/mod.rs

```diff
@@ -2191,13 +2192,26 @@ where
 
         let exec_time = Instant::now();
 
-        // TODO: create StateRootTask with the receiving end of a channel and
-        // pass the sending end of the channel to the state hook.
-        let noop_state_hook = |_result_and_state: &ResultAndState| {};
+        let (state_root_tx, state_root_rx) = std::sync::mpsc::channel();
+
+        let consistent_view = ConsistentDbView::new_with_latest_tip(self.provider.clone())?;
+
+        let input = self
+            .compute_trie_input(consistent_view.clone(), block.parent_hash)
+            .map_err(|e| InsertBlockErrorKindTwo::Other(Box::new(e)))?;
+        let state_root_config =
+            StateRootConfig { consistent_view: consistent_view.clone(), input: Arc::new(input) };
+        let receiver_stream = StdReceiverStream::new(state_root_rx);
+        let state_root_task = StateRootTask::new(state_root_config, receiver_stream);
+        let state_root_handle = state_root_task.spawn();
+        let state_hook = move |result_and_state: &ResultAndState| {
+            let _ = state_root_tx.send(result_and_state.state.clone());
+        };
```

Comment:
is this expected to instantiate a state root task each time, or is it expected that we spawn it once?

Reply:
yep, it is designed to instantiate one task per block. maybe we could get rid of the overhead of creating the task with a background service, similar to what we have for persistence, once everything is working correctly we can measure to compare both approaches